controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -106,7 +108,9 @@ func newToolsDeployment(namespace string, rookImage string) *appsv1.Deployment {
 								},
 							},
 							SecurityContext: &corev1.SecurityContext{
-								Privileged: &privilegedContainer,
+								Privileged:   &privilegedContainer,
+								RunAsNonRoot: &runAsNonRoot,
+								RunAsUser:    &runAsUser,
 							},
 							VolumeMounts: []corev1.VolumeMount{
 								{Name: "dev", MountPath: "/dev"},
```

Comment:
You'll want to remove all the volume mounts except mon-endpoint-volume. See the [toolbox spec](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/toolbox.yaml) we are using upstream that doesn't have the mounts and isn't privileged. To be safe also add the volume for /etc/ceph to match upstream.

Reply:
@travisn I have made the changes according to the comments.

Reply:
@travisn @bindrad If the upstream spec isn't marked as privileged, is there a reason for our d/s deployment to be privileged?

Reply:
@jarrpa If I run the container as an unprivileged user, the container shows an error as the entry point command can't be executed.

Reply:
@jarrpa @travisn It might be that we can do better, but the current changes meet our security concerns and need to go into 4.8. As the deadline for that is near I will suggest to merge this one and introduce any further changes in a future PR

Reply:
@bindrad Did you try removing the security context, or you just found that specifying a non-privileged user failed? Upstream it works fine on openshift to have the default security context. Specifying the user with a privileged context is an improvement over privileged, but would be good to understand why the default security context can't be used.

Reply:
@travisn If I use the default Security context then the container run as a root user

Reply:
It would be non-privileged, which I assumed was the intention. But if the goal is to run as a non-root user, then this looks good.

Reply:
Yes, we need to run the container as non-root user

Reply:
@travisn I do agree that non-privileged will be better, but lets try to accomplish that in another PR

Reply:
Sounds good to me :+1: